WordPress/twentytwentyone
# Move footer widget area

Fixes https://github.com/WordPress/twentytwentyone/issues/472

## Summary
Moves the widget area above the `<footer>` element.
Changes the wrapping `<div>` to `<aside>`.

Updates vertical spacing depending on which elements are active (menu, widgets, or none of them).
Creates a new body class if there are no active widgets in the widget area, to add the extra large top margin to the footer element instead of the widget area.

Makes better use of the CSS variables for text and link color so that these can be changed.

## Relevant technical choices:
See summary.

## Test instructions

This PR can be tested by following these steps:
1. Add widgets to the footer area
2. View the front
3. Check that the widgets are no longer inside the <footer>,
4. Check that the text sizes, link styles, width and spacing is correct across screen sizes.


## Screenshots
Without widgets or menu:
![no-menu-or-widgets](https://user-images.githubusercontent.com/7422055/96075177-c8248d80-0eaa-11eb-8b3a-6d79b463b9c8.png)

Menu only:
![footer-menu](https://user-images.githubusercontent.com/7422055/96075360-28b3ca80-0eab-11eb-9af5-b6bee8cab819.png)

Widgets only:
![footer-widgets](https://user-images.githubusercontent.com/7422055/96075489-6a447580-0eab-11eb-86aa-c45739a647e5.png)

Both menu and widgets:
![footer-menu-and-widgets](https://user-images.githubusercontent.com/7422055/96075561-95c76000-0eab-11eb-9dad-23e6c28418e0.png)



## Quality assurance
